specs/overview.md

```diff
@@ -0,0 +1,205 @@
+search all blocks within the window for transaction batches. This protects against the uncertainty of transaction
+inclusion of L1. This uncertainty is also why we need the sequencing window in the first place: otherwise the sequencer
+could retroactively add blocks to an old epoch, and validators wouldn't know when they can finalize an epoch.
+
+The sequencing window also prevents censorship by the sequencer: deposits made on a given L1 block will be included in
+the L2 chain at worst after `SEQUENCING_WINDOW_SIZE` L1 blocks have passed.
+
+The following diagram describes this relationship, and how L2 blocks are derived from L1 blocks (L1 info transactions
+have been elided):
+
+![Epochs and Sequencing Windows](./assets/sequencer-block-gen.svg)
+
+#### Block Derivation Loop
+
+A sub-component of the rollup node called the _rollup driver_ is actually responsible for performing block derivation.
+The rollup driver is essentially an infinite loop that runs the block derivation function. For each epoch, the block
+derivation function performs the following steps:
+
+1. Downloads deposit and transaction batch data for each block in the sequencing window.
+2. Converts the deposit and transaction batch data into payload attributes for the Engine API.
+3. Submits the payload attributes to the Engine API, where they are converted into blocks and added to the canonical
+chain.
+
+This process is then repeated with incrementing epohcs until the tip of L1 is reached.
+
+### Engine API
+
+The rollup driver doesn't actually create blocks. Instead, it directs the execution engine to do so via the Engine API.
+For each iteration of the block derivation loop described above, the rollup driver will craft a _payload attributes_
+object and send it to the execution engine. The execution engine will then convert the payload attributes object into a
+block, and add it to the chain. The basic sequence the rollup driver is as follows:
+
+1. Call `engine_forkChoiceUpdatedV1` with the payload attributes object. We'll skip over the details of the fork choice
+state parameter for now - just know that one of its fields is the L2 chain's `headBlockHash`, and that it is set to the
+block hash of the tip of the L2 chain. The Engine API returns a payload ID.
+2. Call `engine_getPayloadV1` with the payload ID returned in step 1. The engine API returns a payload object that
+includes a block hash as one of its fields.
+3. Call `engine_executePayloadV1` with the payload returned in step 2.
+4. Call `engine_forkChoiceUpdatedV1` with the fork choice parameter's `headBlockHash` set to the block hash returned in
+step 2. The tip of the L2 chain is now the block created in step 1.
+
+The swimlane diagram below visualizes the process:
+
+![Engine API](./assets/engine.svg)
+
+### Sequencing
+
+> **TODO:** Flesh out this section.
+
+![Sequencing 1](./assets/safe-unsafe-heads.svg)
```

Comment:
This is a dead link

Reply:
Derp, I meant to remove that section altogether. Fixed.